### PR TITLE
LocalVideoTrack improvements

### PR DESCRIPTION
**Pull request by hiroshihorie**
* Unified create capturer logic, since there will be many types of video capturer
* In-app screen capturer (no extension required)
* Capturers can be started/stopped/restarted any time
* Start/Stop capturing for the simplified API to turn off camera/mic indicator
* Switch camera position without restarting track
* Handle rotation for screen capture / video buffer
* macOS desktop capture (main only for now)

### Todo: 
- capturer should report dimensions
- unify interceptor interface for other capturers also
### Thoughts
- `restartTrack` not really required anymore?

### CODE REVIEW

**Review by boks1971 — COMMENTED** [review]
lgtm! But I do not understand all the nuances yet :-) Would be good to get another pair of :eyes:

**Comment by hiroshihorie** [comment]
@boks1971  😁🙏

File: Sources/LiveKit/participant/LocalParticipant.swift [hunk 1]
```diff
@@ -230,35 +232,47 @@ extension LocalParticipant {
         return set(source: .microphone, enabled: enabled)
     }
 
+    public func setScreen(enabled: Bool) -> Promise<LocalTrackPublication?> {
```
Comment (davidliu) [inline Sources/LiveKit/participant/LocalParticipant.swift:235]: change function name to `setScreenShare` just to be fully explicit.

File: Sources/LiveKit/participant/LocalParticipant.swift [hunk 2]
```diff
@@ -230,35 +232,47 @@ extension LocalParticipant {
...
+                return publishVideoTrack(track: localTrack).then { publication in return publication }
+            } else if source == .microphone {
                 let localTrack = LocalAudioTrack.createTrack(name: "")
                 return publishAudioTrack(track: localTrack).then { publication in return publication }
+            } else if source == .screenShareVideo {
+
+                var localTrack: LocalVideoTrack?
+
+                #if !os(macOS)
+                if #available(macOS 11.0, *) {
```
Comment (davidliu) [inline Sources/LiveKit/participant/LocalParticipant.swift:266]: This looks a little odd? Tests if it's not macOS, but then checks if it's macOS 11.0 available? Is this supposed to be for iOS?

File: Sources/LiveKit/participant/Participant.swift [hunk 3]
```diff
@@ -75,6 +75,10 @@ extension Participant {
         !(getTrackPublication(source: .microphone)?.muted ?? true)
     }
 
+    public func isScreenEnabled() -> Bool {
```
Comment (davidliu) [inline Sources/LiveKit/participant/Participant.swift:78]: `isScreenShareEnabled`

File: Sources/LiveKit/track/LocalVideoTrack.swift [hunk 4]
```diff
@@ -1,179 +1,74 @@
...
-        }
-        if fps < minFps || fps > maxFps {
-            throw TrackError.mediaError("requested framerate is unsupported (\(minFps)-\(maxFps))")
-        }
-
-        logger.info("starting capture with \(device), format: \(selectedFormat), fps: \(fps)")
-        capturer.startCapture(with: device, format: selectedFormat, fps: Int(fps))
-
-        return (capturer, output, selectedDimension)
+    public func restartTrack(options: LocalVideoTrackOptions = LocalVideoTrackOptions()) {
```
Comment (davidliu) [inline Sources/LiveKit/track/LocalVideoTrack.swift:37]: Empty function?

**Review by davidliu — APPROVED** [review]
LGTM

**Comment by hiroshihorie** [comment]
Thanks alot onichans 😁🙏

File: Sources/LiveKit/participant/LocalParticipant.swift (hunk 2 above)
Comment (hiroshihorie) [reply Sources/LiveKit/participant/LocalParticipant.swift:275]: thanks for finding this!! yea thats weird 😅

**Pull request merged**